src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java

```diff
@@ -72,89 +71,78 @@ public OpenCompositeOrderPieceService(PurchaseOrderStorageService purchaseOrderS
    * @return void future
    */
   public Future<List<Piece>> handlePieces(CompositePoLine compPOL, String titleId, List<Piece> expectedPiecesWithItem,
-                                                     boolean isInstanceMatchingDisabled, RequestContext requestContext) {
-    logger.debug("Get pieces by poLine ID");
+                                          boolean isInstanceMatchingDisabled, RequestContext requestContext) {
+    logger.debug("handlePieces:: Get pieces by poLine ID - {}", compPOL.getId());
     return openCompositeOrderHolderBuilder.buildHolder(compPOL, titleId, expectedPiecesWithItem, requestContext)
       .compose(holder -> {
-        if (CollectionUtils.isNotEmpty(holder.getPiecesWithLocationToProcess()) &&
-                       (holder.getPiecesWithChangedLocation().size() == holder.getPiecesWithLocationToProcess().size())) {
-          return updatePieces(holder, requestContext);
-        } else {
-          return createPieces(holder, isInstanceMatchingDisabled, requestContext);
+        var piecesChangedLocation = holder.getPiecesWithChangedLocation();
+        if (CollectionUtils.isEmpty(piecesChangedLocation) || piecesChangedLocation.size() != holder.getPiecesWithLocationToProcess().size()) {
+          return purchaseOrderStorageService.getCompositeOrderById(compPOL.getPurchaseOrderId(), requestContext)
```

Comment:
good refactoring with reducing number of calls to get order for each piece.
piecesChangedLocation this name tells that it is location object, can be renamed to piecesWithChangedLocation